.cbmc-batch/jobs/aws_cryptosdk_enc_ctx_clear/aws_cryptosdk_enc_ctx_clear_harness.c

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <aws/common/hash_table.h>
+#include <aws/common/private/hash_table_impl.h>
+#include <aws/cryptosdk/enc_ctx.h>
+#include <proof_helpers/make_common_data_structures.h>
+#include <proof_helpers/proof_allocators.h>
+#include <proof_helpers/utils.h>
+
+void aws_cryptosdk_enc_ctx_clear_harness() {
+    struct aws_hash_table map;
+    ensure_allocated_hash_table(&map, MAX_TABLE_SIZE);
+    __CPROVER_assume(aws_hash_table_is_valid(&map));
+    ensure_hash_table_has_valid_destroy_functions(&map);
+
+    size_t empty_slot_idx;
+    __CPROVER_assume(aws_hash_table_has_an_empty_slot(&map, &empty_slot_idx));
```

Comment:
It's not clear to me why you need to assume this for the proof to work

Reply:
In general, this is an important property of the hash-table, that ensures that certain hash-table functions terminate: they continue until they find the first empty slot, so if there is no empty slot, they never terminate.  This is a property that is ensured by the hash-table contract, so its safe to assume.  What I'd really like would be a way to ensure this in the ```aws_hash_table_is_valid()``` function, but I don't see an easy way that doesn't involve an expensive loop.

Reply:
That said, I think we're not calling any of the dangerous hash functions, so this may be an unneeded assumption.